### DISCUSSION TITLE
Git status icon like (staged, unstaged, renamed, . . . ) does not appear in nvim-tree?

### DISCUSSION THREAD

**Discussion by reynaldlamury** [kyazdani42/nvim-web-devicons#263] [Q&A]
I have **nvim-tree/nvim-tree** and **nvim-tree/nvim-web-devicons** plugins installed on my nvim lua based config. Git status icon does not appear. The status icon is usually located at the left handside of the file/folder.

Here is the **snapshot** that can show how It should look like if the icon appears:

![git status- nvim tree](https://user-images.githubusercontent.com/51454984/235384676-7759e6b9-2754-4b66-b817-58a0efe4e948.png)

I dont't know why it does not work. I have set the option `renderer = {icons = {glyphs = {git = {unstaged= "U", ...}}}}` . Is there something wrong with this ?

Here is the **packer manager file** that contains all plugins. In case it might has something to do with the way I order the plugin most particularly the "nvim-tree/nvim-web-devicons" plugin.

**packer.lua**
```
vim.cmd([[packadd packer.nvim]])

return require("packer").startup(function(use)
    --[[ rest of the plugins ]]

    use({ --- lualine
        "nvim-lualine/lualine.nvim",
        requires = { "nvim-tree/nvim-web-devicons" },
    })

    use({ -- LSPSaga
        "glepnir/lspsaga.nvim",
        opt = true,
        branch = "main",
        event = "LspAttach",
        config = function()
            require("lspsaga").setup({})
        end,
        requires = {
            { "nvim-tree/nvim-web-devicons" },
            --Please make sure you install markdown and markdown_inline parser
            { "nvim-treesitter/nvim-treesitter" },
        },
    })

    --[[ rest of the plugins ]]
end)
```

Or There might be a conflict between nvim-tree plugin and other plugins. But I'm not sure where the problem is. Any Idea ?

**Comment by gegoune** [comment]
You haven't shown relevant nvim-tree setup configuration, or what it looks like when it doesn't work.

This works for me 

```lua
  renderer = {
    icons = {
      git_placement = 'signcolumn',
      glyphs = {
        git = {
          unstaged = 'M',
          staged = 'S',
          unmerged = 'U',
          renamed = 'R',
          untracked = '?',
          deleted = 'D',
          ignored = 'I',
        },
      },
    },
  },
```

**Reply by reynaldlamury** [reply]
Mine looks like this:

**nvim-tree.lua**
```
-- following options are the default
-- each of these are documented in `:help nvim-tree.OPTION_NAME`

local status_ok, nvim_tree = pcall(require, "nvim-tree")
if not status_ok then
	return
end

--[[ local config_status_ok, nvim_tree_config = pcall(require, "nvim-tree.config") ]]
--[[ if not config_status_ok then ]]
--[[ 	return ]]
--[[ end ]]

--[[ local tree_cb = nvim_tree_config.nvim_tree_callback ]]

nvim_tree.setup({
	filters = {
		dotfiles = false,
		git_clean = false,
		no_buffer = false,
		custom = {
			"node_modules", -- filter out node_modules directory
			".git", -- filter out .git directory
		},
		exclude = {},
	},
	disable_netrw = true,
	hijack_netrw = true,
	--open_on_setup = false,
	open_on_tab = false,
	hijack_cursor = false,
	update_cwd = true,
	hijack_directories = {
		enable = true,
		auto_open = true,
	},
	diagnostics = {
		enable = true,
		icons = {
			hint = "",
			info = "",
			warning = "",
			error = "",
		},
	},
	update_focused_file = {
		enable = true,
		update_cwd = true,
		ignore_list = {},
	},
	git = {
		enable = true,
		ignore = true,
		timeout = 500,
	},
	view = {
		width = 30,
		--height = 30,
		hide_root_folder = false,
		side = "left",
		--auto_resize = true,
		mappings = {
			custom_only = false,
		},
		number = false,
		relativenumber = false,
	},
	actions = {
		--quit_on_open = true,
		--window_picker = { enable = true },
	},
	renderer = {
		highlight_git = true,
		root_folder_modifier = ":t",
		icons = {
			show = {
				file = true,
				folder = true,
				folder_arrow = true,
				git = true,
			},
			glyphs = {
				default = "",
				symlink = "",
				git = {
					unstaged = "U",
					staged = "S",
					unmerged = "UM",
					renamed = "R",
					deleted = "D",
					untracked = "",
					ignored = "I",
				},
				folder = {
					default = "",
					open = "",
					empty = "",
					empty_open = "",
					symlink = "",
				},
			},
		},
	},
})

```

**Reply by gegoune** [reply]
Can't replicate it, sorry. Please open new issue with reproducible minimal configuration and steps in nvim-tree repository as it seem unrelated to nvim-web-devicons (you are setting git glyphs to ascii characters).

**Reply by gegoune** [reply]
Also make sure you have proper font installed and configured correctly.

**Reply by reynaldlamury** [reply]
I've used NerdFont. I set some of the git status icon a text-based symbol such as for `unstaged = "U"`, for `renamed = "R"` . But They still don't appear. It's weird, All the icons are showing up except the git symbol icon @gegoune

**Reply by reynaldlamury** [reply]
Sorry but I don't understand what you mean by 'create reproducible minimal configuration and steps in nvim-tree repository' ? Can you walk me through the steps ? @gegoune

**Reply by alex-courtis** [reply]
Please see [Clean Room Replication](https://github.com/nvim-tree/nvim-tree.lua/wiki/Troubleshooting#clean-room-replication)

nvim-tree will use nvim-web-devicons
